### PR TITLE
SSCS-11706 - Error when attempting to upload hearing recording onto a List assist case

### PR DESCRIPTION
**Pull request by hajam09**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Review by jarekPierchala — CHANGES_REQUESTED** [review]

File: src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/uploadhearingrecording/UploadHearingRecordingAboutToStartHandlerTest.java [hunk 1]
```diff
@@ -137,6 +137,51 @@ public void givenHearingsListHasPastDate_ReturnList() {
...
+        hearingList.add(Hearing.builder().value(hearingDetails4).build());
+        hearingList.add(Hearing.builder().value(hearingDetails5).build());
+
+        sscsCaseData.setHearings(unmodifiableList(hearingList));
+        final PreSubmitCallbackResponse<SscsCaseData> response = handler.handle(ABOUT_TO_START, callback, USER_AUTHORISATION);
+        assertEquals(0, response.getErrors().size());
+
+        List<DynamicListItem> selectHearingDetails = response.getData().getSscsHearingRecordingCaseData().getSelectHearingDetails().getListItems();
+        assertEquals(hearingList.size(), selectHearingDetails.size());
+        selectHearingDetails.forEach(o -> assertEquals("venue name 09:00:00 06 Jun 2021", o.getLabel()));
```
Comment (amoslv1) [inline src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/uploadhearingrecording/UploadHearingRecordingAboutToStartHandlerTest.java:182]: You can avoid using a forEach
```suggestion
        assertTrue(selectHearingDetails.stream()
    .allMatch(o -> "venue name 09:00:00 06 Jun 2021".equals(o.getLabel())));
```

**Review by amoslv1 — CHANGES_REQUESTED** [review]
small optimisation in test

**Review by EliandroGV1 — DISMISSED** [review]

**Review by amoslv1 — DISMISSED** [review]

**Review by jarekPierchala — APPROVED** [review]

**Review by nickhill111 — APPROVED** [review]

**Review by EliandroGV1 — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/uploadhearingrecording/UploadHearingRecordingAboutToStartHandler.java [hunk 2]
```diff
@@ -75,14 +73,33 @@ public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType,
...
-            + LocalDate.parse(hearing.getValue().getHearingDate(), formatter).format(resultFormatter);
+        String venueName = hearing.getValue().getVenue().getName();
+        String hearingTime = checkHearingTime(hearing.getValue().getTime());
+        String hearingDate = LocalDate.parse(hearing.getValue().getHearingDate(), formatter).format(resultFormatter);
+        String returnValue = venueName;
+
+        if (!hearingTime.equals("")) {
+            returnValue += " " + hearingTime;
+        }
+        return returnValue + " " + hearingDate;
```
Comment (jarekPierchala) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/uploadhearingrecording/UploadHearingRecordingAboutToStartHandler.java:84]:
```suggestion
      
        return !hearingTime.equals("") ? returnValue + " " + hearingTime : returnValue + " " + hearingDate;
```

**Review by jarekPierchala — CHANGES_REQUESTED** [review]

**Review by AndreiTrushkov — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/uploadhearingrecording/UploadHearingRecordingAboutToStartHandler.java (hunk 2 above)
Comment (hajam09) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/uploadhearingrecording/UploadHearingRecordingAboutToStartHandler.java:84]: Unfortunately, this is breaks my test.

**Review by jarekPierchala — APPROVED** [review]

**Pull request merged**